pkg/tracing/tracing.go

```diff
@@ -0,0 +1,132 @@
+// Copyright 2017 Google Inc.
```

Comment:
Can you get code coverage to 100% in this file and in metadata.go? You're pretty close, just a few more edge cases to handle.